src/main/java/org/folio/service/pieces/flows/delete/PieceDeleteFlowManager.java

```diff
@@ -62,89 +45,39 @@ public PieceDeleteFlowManager(PieceStorageService pieceStorageService,
-        .map(v -> {
-          if (CollectionUtils.isNotEmpty(combinedErrors)) {
-            Errors errors = new Errors().withErrors(combinedErrors).withTotalRecords(combinedErrors.size());
-            logger.error("Validation error : " + JsonObject.mapFrom(errors).encodePrettily());
-            throw new HttpException(RestConstants.VALIDATION_ERROR, errors);
-          }
-          return null;
-        })
-      .mapEmpty();
-    }
-    return Future.succeededFuture();
-  }
-
-  private Future<Pair<String, String>> processInventory(PieceDeletionHolder holder, RequestContext requestContext) {
-    return deleteItem(holder, requestContext)
-               .compose(aVoid -> {
-                 if (holder.isDeleteHolding()) {
-                   return pieceUpdateInventoryService.deleteHoldingConnectedToPiece(holder.getPieceToDelete(), requestContext);
-                 }
-                 return Future.succeededFuture();
-               });
-  }
-
-  protected Future<Void> updatePoLine(PieceDeletionHolder holder, RequestContext requestContext) {
-    if (!Boolean.TRUE.equals(holder.getOriginPoLine().getIsPackage()) && !Boolean.TRUE.equals(holder.getOriginPoLine().getCheckinItems())) {
-      return  pieceDeleteFlowPoLineService.updatePoLine(holder, requestContext);
+    var piece = holder.getPieceToDelete();
+    if (piece.getItemId() == null) {
+      return Future.succeededFuture();
     }
-    return Future.succeededFuture();
-  }
 
-  private Future<Void> deleteItem(PieceDeletionHolder holder, RequestContext requestContext) {
-    Piece piece = holder.getPieceToDelete();
-    if (piece.getItemId() != null) {
-      return getOnOrderItemForPiece(piece, requestContext).compose(item -> {
-        if (item != null) {
-          return inventoryItemManager.deleteItem(piece.getItemId(), true, requestContext);
+    return circulationRequestsRetriever.getNumberOfRequestsByItemId(piece.getItemId(), requestContext)
+      .compose(totalRequests -> {
+        if (totalRequests != null && totalRequests > 0) {
+          logger.error("isDeletePieceRequestValid:: {} Request(s) were found for the given item {} when deleting piece {}",
+            totalRequests, piece.getItemId(), piece.getId());
+          throw new HttpException(RestConstants.VALIDATION_ERROR, ErrorCodes.REQUEST_FOUND.toError());
         }
         return Future.succeededFuture();
-      });
-    }
-    return Future.succeededFuture();
+      })
+      .mapEmpty();
   }
 
-  private boolean isItemWithStatus(JsonObject item, String status) {
-    return Optional.ofNullable(item).map(itemP -> item.getJsonObject(ITEM_STATUS))
-      .filter(itemStatus -> status.equalsIgnoreCase(itemStatus.getString(ITEM_STATUS_NAME)))
-      .isPresent();
+  protected Future<Void> updatePoLine(PieceDeletionHolder holder, RequestContext requestContext) {
+    var comPOL = holder.getOriginPoLine();
+    return BooleanUtils.isTrue(comPOL.getIsPackage()) || BooleanUtils.isTrue(comPOL.getCheckinItems())
```

Comment:
to be consistent, lets use Boolean.FALSE or TRUE equals

Reply:
If you are doing, please change in other places of pr

Reply:
Got rid of it in all places